addons/account/models/account_account.py

```diff
@@ -346,20 +402,160 @@ def _check_account_is_bank_journal_bank_account(self):
         if self._cr.fetchone():
             raise ValidationError(_("You cannot change the type of an account set as Bank Account on a journal to Receivable or Payable."))
 
+    def _compute_name(self):
+        for company in (self.env.company.parent_ids | self.company_ids):
+            values = self.env['ir.property'].sudo().with_company(company)._get_multi('name', 'account.account', self.ids)
+            for record in self:
+                if (company in self.env.company.parent_ids | record.company_ids):
```

Comment:
```suggestion
        relevant_companies = self.env.company.parent_ids | self.company_ids
        for company in relevant_companies:
            values = self.env['ir.property'].sudo().with_company(company)._get_multi('name', 'account.account', self.ids)
            for record in self:
                if company in relevant_companies:
```

Reply:
Not quite :smile: 
The reason for the double-check is that the relevant companies depends on the record.
E.g. record 1 has company_ids 1, 2
record 2 has company_ids 3, 4
Active company is company 5, which is not a child of 1,2,3 or 4.
For record 1 we want to see the name for company 1 (if set) or company 2
For record 2 we want to see the name for company 3 (if set) or company 4